web/php/Actions/WikiPageAction.php

```diff
@@ -1407,8 +1414,17 @@ public function saveTagsEvent($runData)
 
         WDPermissionManager::instance()->hasPagePermission('edit', $user, $category, $page);
 
-        if (strlen8($tags)>256) {
-            throw new ProcessException(_('"Tags" field too long.'), "form_error");
+        $pageTagsArray = explode(' ', $tags);
+
+        $allowedTagsList = AllowedTags::getAllowedTags($siteId);
+        $allowedTagsList = explode(' ', $allowedTagsList);
+
+       if($enableAllowedTags == true) {
+            foreach ($pageTagsArray as $tag) {
+                if(!in_array($tag, $allowedTagsList)) {
+                    throw new ProcessException(_('Invalid tags.'), "form_error");
```

Comment:
I would give it a more specific error message, like "Tag XYZ is not valid for this site."